### PR TITLE
Add prefect-dask recipe

### PR DESCRIPTION
**Pull request by ahuang11**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by desertaxle** [comment]
Confirming that I am willing to be listed as a recipe maintainer

**Comment by madkinsz** [comment]
Confirming that I am willing to be listed as a recipe maintainer

**Comment by ahuang11** [comment]
@conda-forge-admin, please ping conda-forge/help-python

File: recipes/prefect-dask/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - pip
+    - python >=3.6
+  run:
+    - distributed >=2022.2.0
+    - prefect >=2.3.0
+    - python >=3.6
+
+test:
+  imports:
+    - prefect_dask
```
Comment (dbast) [inline recipes/prefect-dask/meta.yaml:28]:
```suggestion
  imports:
    - prefect_dask
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/prefect-dask/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - pip
+    - python >=3.6
+  run:
+    - distributed >=2022.2.0
+    - prefect >=2.3.0
+    - python >=3.6
```
Comment (dbast) [inline recipes/prefect-dask/meta.yaml:24]: According to setup.py: `python_requires=">=3.7"`, see https://github.com/PrefectHQ/prefect-dask/blob/v0.2.0.post1/setup.py#L27

```suggestion
    - python >=3.7
  run:
    - distributed >=2022.2.0
    - prefect >=2.3.0
    - python >=3.7
```

**Comment by dbast** [comment]
Recipe looks good, except the last inline comment.

**Review by dbast — APPROVED** [review]

**Comment by ahuang11** [comment]
I just tried installing prefect-dask through mamba.
```
Encountered problems while solving:
  - nothing provides asgi-lifespan >=1.0 needed by prefect-2.3.0-pyhd8ed1ab_0
```

However, when I install prefect through conda, it works successfully, any ideas why?

**Comment by dbast** [comment]
@ahuang11 That should make it available https://github.com/conda-forge/asgi-lifespan-feedstock/pull/8

**Comment by dbast** [comment]
This is also required for osx-arm64 https://github.com/conda-forge/coolname-feedstock/pull/8

**Comment by dbast** [comment]
Last missing thing is https://github.com/conda-forge/griffe-feedstock ... which is noarch but only for newer versions, prefect 2.3.0 needs griffe >=0.20.0, <0.22.0

**Comment by ahuang11** [comment]
To clarify, do I need to update the recipe to include coolname + griffe, or will these PRs automatically make it work?

**Comment by dbast** [comment]
So `mamba create --name test prefect` works on apple M1, but installs the prefect 1.X version. Forcing prefect 2 via `mamba create --name test prefect==2.3.0` or `mamba create --name test prefect-dask` (prefect-dask depends on prefect 2) reveals that several dependencies are missing on Apple M1 (osx-arm64):
* asgi-lifespan: fixed by making it noarch -> available
* coolname: fixed proposed by makign ti noarch -> merge review/pending
* griffe: conda-forge current version is 0.22.0, but prefect needs <0.22.0.... older versions are not noarch... so two options: 1. prefect 2 lifts its griffe dependencies (version maybe only relevant for doc creation but not runtime) 2. an older griffe version (0.21.0) is made available on as noarch on conda-forge.

**Comment by dbast** [comment]
Older griffe version should be available soon as noarch, see https://github.com/conda-forge/griffe-feedstock/pull/35. The coolname PR is hopefully merged soon.

**Comment by dbast** [comment]
All PRs got merged, it is now installable on Apple M1 machines: `CONDA_SUBDIR=osx-arm64 mamba create --name test prefect-dask`

**Comment by ahuang11** [comment]
Thank you so much for helping us get it installable!

**Pull request merged**